gui/default/syncthing/folder/editFolderModalView.html

```diff
@@ -11,11 +11,22 @@ <h4 class="modal-title">
         <form role="form" name="folderEditor">
           <div class="row">
             <div class="col-md-12">
+              <div class="form-group" ng-class="{'has-error': folderEditor.folderLabel.$invalid && folderEditor.folderLabel.$dirty}">
+                <label for="folderLabel"><span translate>Folder Label</span></label>
+                <input name="folderLabel" id="folderLabel" class="form-control" type="text" ng-model="currentFolder.label" ng-if="editingExisting"/>
+                <input name="folderLabel" id="folderLabel" class="form-control" type="text" ng-model="currentFolder.label" ng-if="!editingExisting &&  currentFolder.viewFlags.importFromOtherDevice" value="{{currentFolder.label}}"/>
```

Comment:
OK I get it. I think this gets simplified if you prefill the auto generated folder ID before bringing up the dialog (instead of using the folder label changed callback). I like your viewFlags things though. Maybe that could replace the `editingExisting` flag so that you instead have something like `viewFlags.canEditFolderID` and `viewFlags.canEditFolderPath`. When editing existing folders, both are false. When adding a new folder manually, both are true. When adding a folder from a sharing request, `canEditFolderID` is false but `canEditFolderPath` is true?